connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -941,6 +942,36 @@ message ApplyInPandasWithState {
   string timeout_conf = 7;
 }
 
+message CommonInlineUserDefinedTableFunction {
+  // (Required) Name of the user-defined table function.
+  string function_name = 1;
+
+  // (Optional) Whether the user-defined table function is deterministic.
+  bool deterministic = 2;
+
+  // (Optional) Function input arguments. Empty arguments are allowed.
+  repeated Expression arguments = 3;
+
+  // (Required) Type of the user-defined table function.
+  oneof function {
+    PythonUDTF python_udtf = 4;
```

Comment:
Out of curiosity will there be UDTFs in other languages?

Reply:
Currently only Python but more UDTFs might be supported in the future.